TreeMaker/src/TreeMaker.cc

```diff
@@ -48,39 +48,77 @@ TreeMaker::TreeMaker(const edm::ParameterSet& iConfig) :
 	treeName = iConfig.getParameter<string>("TreeName");
 	doLorentz = iConfig.getParameter<bool>("doLorentz");
 	sortBranches = iConfig.getParameter<bool>("sortBranches");
-	//loop over all var type names to initialize TreeObjects
-	set<string> nameSet;
+	debugTitles = iConfig.getParameter<bool>("debugTitles");
+
+	// parse the TitleMap
 	stringstream skipMessage;
 	stringstream message;
+	set<string> titleSet;
+	vector<string> titleVector = iConfig.getParameter< vector<string> >("TitleMap");
+	if(titleVector.size()%2 != 0) {
+		cms::Exception ex("LogicError", "The vector containing the InputTags and titles is odd, indicating that there is not a 1-to-1 pairing");
```

Comment:
"is odd" -> "has an odd number of entries"

Reply:
Fixed